versioned_docs/version-2.0/tutorials/Developers/swe-benchmark.mdx

```diff
@@ -0,0 +1,287 @@
+```
+
+### Editing the 'version' column
+
+<div class="alert alert-block alert-danger">
+  This is a very important step! If you skip, the rest of the code WILL NOT
+  WORK!
+</div>
+
+The `version` column contains all string values but all are in float format so they get converted to floats when you upload the CSV to create a LangSmith dataset. Although you can convert the values to strings during your experiments, the issue arises with values like `"0.10"`. When getting converted to a float, you get the value `0.1`, which would become `"0.1"` if you converted it to a string - causing a key error during execution of your proposed patch.
+
+In order to fix this, we need LangSmith to stop trying to convert the `version` column to floats. In order to do this, we can just append a string prefix to each of them that is not float compatible. We then need to split on this prefix when doing evaluation to get the actual `version` value. The prefix we choose here is the string `"version:"`.
+
+<div class="alert alert-block alert-info">
+  The ability to select column types when uploading a CSV to LangSmith will be
+  added in the future to avoid having to use this workaround.
+</div>
+
+```python
+df['version'] = df['version'].apply(lambda x: f"version:{x}")
+```
+
+## Upload the data to LangSmith
+
+### Save to CSV
+
+To upload the data to LangSmith, we first need to save it to a CSV, which we can do using the `to_csv` function provided by pandas. Make sure to save this file somewhere that is easily accesible to you.
+
+```python
+df.to_csv("./../swe-benchmark.csv",index=False)
+```
+
+### Upload CSV to LangSmith
+
+We are now ready to upload the CSV to LangSmith. Once you are on the LangSmith website (smith.langchain.com), go to the `Datasets & Testing` tab on the left side navigation bar, and then click the `+ New Dataset` button in the top right corner.
+
+Then click the `Upload CSV` button on the top, and select the CSV file you saved in the previous step. You can then give your dataset a name and description.
+
+Next, select `Key-Value` as the dataset type. Lastly head to the `Create Schema` section and add ALL OF THE KEYS as `Input fields`. There are no `Output fields` in this example because our evaluator is not comparing against a reference, but instead will run the output of our experiments in docker containers to ensure that the code actually solves the PR issue.
+
+Once you have populated the `Input fields` (and left the `Output fields` empty!) you can click the blue `Create` button in the top right corner, and your dataset will be created!
+
+### Create dataset split for quicker testing
+
+Since running the SWE-benchmark evaluator takes a long time when run on all examples, you can create a "test" split for quickly testing the evaluator and your code. This video shows you how to do so (to get to the starting page of the video, just click on your dataset created above and go to the `Examples` tab):
```

Comment:
link to here: https://docs.smith.langchain.com/how_to_guides/datasets/manage_datasets_in_application#create-and-manage-dataset-splits